app/user_views.py

```diff
@@ -1,42 +1,75 @@
-from flask_login import logout_user, current_user, login_required
-from flask import render_template, url_for, redirect, flash, request
+from flask_login import current_user, login_required
+from flask import render_template, url_for, redirect, flash, request, abort
 from flask import current_app as app
 from sqlalchemy import insert
-from .models import db, Rehearsal
-from .forms import RehearsalForm
+from .models import db, Rehearsal, User
+from .forms import RehearsalForm, OrderForm
 import datetime as dt
+from functools import wraps
 
 
-@app.route('/logout')
-def logout():
-    logout_user()
-    return redirect(url_for('index'))
+def check_rehearsal_user(func):
```

Comment:
Very nice!

Once you're writing decorators, I'm pretty sure you're intermediate, not novice. 🙌🏽💪🏽

Reply:
🥳